node/uhpo/src/payout_update.rs

```diff
@@ -0,0 +1,282 @@
+        let mut builder = TransactionBuilder::new().add_input(coinbase_tx.compute_txid(), 0);
+
+        // Add previous update transaction input if it exists
+        // previous update tx would be None if and only if this is the first update
+        let prev_update_txout = if let Some(tx) = prev_update_tx {
+            builder = builder.add_input(tx.compute_txid(), 0);
+            Some(tx.output[0].clone())
+        } else {
+            None
+        };
+
+        // Calculate total amount from coinbase and previous update
+        let total_amount = prev_update_txout.as_ref().map_or_else(
+            || coinbase_txout.value,
+            |txout| coinbase_txout.value + txout.value,
+        );
+
+        let transaction = builder
+            .add_output(next_out_address, total_amount - projected_fee)
+            .build();
+
+        Ok(PayoutUpdate {
+            transaction,
+            coinbase_txout,
+            prev_update_txout,
+        })
+    }
+
+    /// Adds a signature for the coinbase input.
+    ///
+    /// # Arguments
+    /// * `private_key` - The private key for signing
+    /// * `tweak` - Optional tweak for the private key
+    /// * `secp` - The Secp256k1 context
+    ///
+    /// # Returns
+    /// Result indicating success or an UhpoError
+    pub fn add_coinbase_sig(
+        &mut self,
+        private_key: SecretKey,
+        tweak: Option<&Scalar>,
+        secp: &Secp256k1<All>,
+    ) -> Result<(), UhpoError> {
+        let prevouts = match &self.prev_update_txout {
+            Some(prev_update_txout) => vec![&self.coinbase_txout, prev_update_txout],
+            None => vec![&self.coinbase_txout],
+        };
+
+        add_signature(
+            &mut self.transaction,
+            0, // coinbase input idx
+            &prevouts,
+            private_key,
+            tweak,
+            secp,
+        )
+    }
+
+    /// Adds a signature for the previous update input.
+    ///
+    /// # Arguments
+    /// * `private_key` - The private key for signing
+    /// * `tweak` - Optional tweak for the private key
+    /// * `secp` - The Secp256k1 context
+    ///
+    /// # Returns
+    /// Result indicating success or an UhpoError
+    pub fn add_prev_update_sig(
+        &mut self,
+        private_key: SecretKey,
+        tweak: Option<&Scalar>,
+        secp: &Secp256k1<All>,
+    ) -> Result<(), UhpoError> {
+        let prev_update_txout = self
+            .prev_update_txout
+            .as_ref()
+            .ok_or(UhpoError::NoPrevUpdateTxOut)?;
+        let prevouts = vec![&self.coinbase_txout, prev_update_txout];
+
+        add_signature(
+            &mut self.transaction,
+            1,
+            &prevouts,
+            private_key,
+            tweak,
+            secp,
+        )
+    }
+
+    pub fn build(self) -> Transaction {
+        self.transaction
+    }
+}
+
+// unit tests
+#[cfg(test)]
+mod tests {
```

Comment:
Let's add a test to see we handle errors properly from `add_signature`. If there is a match on the error, we should have one test case for each.